### PR TITLE
nixos/uptime-kuma: add ping in path

### PR DESCRIPTION
**Pull request by JulienMalka**
###### Description of changes

Version 1.19.4 of uptime-kuma introduced a new method for the ping monitor, using the installed package instead of a library. Added support for 1.19.4+ by adding ping to the path.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.05 Release Notes (or backporting 22.11 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2305-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by JulienMalka** [comment]
Tested uptime-kuma 1.19.6 with this module update and can confirm it works.

**Comment by JulienMalka** [comment]
I think this should be backported because users using 21.11 for their nixos channel and unstable.uptime-kuma as the package will see their setup broken.

File: nixos/modules/services/monitoring/uptime-kuma.nix [hunk 1]
```diff
@@ -48,7 +48,7 @@ in
       after = [ "network.target" ];
       wantedBy = [ "multi-user.target" ];
       environment = cfg.settings;
-      path = lib.mkIf cfg.appriseSupport (with pkgs; [ apprise ]);
+      path = with pkgs; [ unixtools.ping ] ++ (if cfg.appriseSupport then [ apprise ] else [ ]);
```
Comment (RaitoBezarius) [inline nixos/modules/services/monitoring/uptime-kuma.nix:51]:
```suggestion
      path = with pkgs; [ unixtools.ping ] ++ (lib.optional cfg.appriseSupport apprise);
```

**Review by RaitoBezarius — APPROVED** [review]
Minor change and should be fine IMHO.

File: nixos/modules/services/monitoring/uptime-kuma.nix (hunk 1 above)
Comment (JulienMalka) [reply nixos/modules/services/monitoring/uptime-kuma.nix]: Thanks, done :)

File: nixos/modules/services/monitoring/uptime-kuma.nix [hunk 2]
```diff
@@ -48,7 +48,7 @@ in
       after = [ "network.target" ];
       wantedBy = [ "multi-user.target" ];
       environment = cfg.settings;
-      path = lib.mkIf cfg.appriseSupport (with pkgs; [ apprise ]);
+      path = with pkgs; [ unixtools.ping ] ++ (lib.optional cfg.appriseSupport apprise);
```
Comment (SuperSandro2000) [inline nixos/modules/services/monitoring/uptime-kuma.nix:51]:
```suggestion
      path = with pkgs; [ unixtools.ping ] ++ lib.optional cfg.appriseSupport apprise;
```

**Comment by SuperSandro2000** [comment]
We don't need to backport this because we are not going to backport the update for uptime-kuma.

**Pull request merged**